clap_complete/src/dynamic/completer.rs

```diff
@@ -655,6 +654,21 @@ fn parse_positional<'a>(
     }
 }
 
+/// Parse optional flag argument. Return new state
+fn parse_opt(arg: &clap::Arg, count: usize) -> ParseState<'_> {
```

Comment:
nit: in the future, please put refactors into their own commits before feature work